src/test/scala/chiselTests/PrintableSpec.scala

```diff
@@ -261,4 +261,208 @@ class PrintableSpec extends AnyFlatSpec with Matchers {
       """printf(clock, UInt<1>("h1"), "adieu AnonymousBundle(foo -> %d, bar -> %d)", myBun.foo, myBun.bar) : farewell"""
     )
   }
+
+  // Unit tests for cf
+  it should "handle Printable and String concatination with cf format specifier" in {
+    class MyModule extends BasicTester {
+      printf(cf"First " + PString("Second ") + "Third")
+    }
+    val firrtl = ChiselStage.emitChirrtl(new MyModule)
+    getPrintfs(firrtl) match {
+      case Seq(Printf("First Second Third", Seq())) =>
+      case e                                        => fail()
+    }
+  }
+  it should "call toString on non-Printable objects with cf format specifier" in {
+    class MyModule extends BasicTester {
+      val myInt = 1234
+      printf(cf"myInt = $myInt")
+    }
+    val firrtl = ChiselStage.emitChirrtl(new MyModule)
+    getPrintfs(firrtl) match {
+      case Seq(Printf("myInt = 1234", Seq())) =>
+      case e                                  => fail()
+    }
+  }
+  it should "generate proper printf for simple Decimal printing with cf format specifier" in {
+    class MyModule extends BasicTester {
+      val myWire = WireDefault(1234.U)
+      printf(cf"myWire = ${Decimal(myWire)}")
```

Comment:
```suggestion
      printf(cf"myWire = $myWire%d")
```
The purpose of `cf` is to support this style 🙂

Reply:
Also need to test things like `$mywire%x`, and I think that the the current implementation doesn't work if you have normal Strings after the modifier, eg. `printf(cf"myWire = $myWire%d!!!\n)`

Reply:
As discussed - these set of tests were copied over from "p" to ensure cf is indeed a superset.

Reply:
Working on refactoring / adding more tests - please review this file later.

Reply:
Continuing to add more tests - but at least this basic scenario seems to work - including strings right after %x of the wire. @jackkoenig  will continue testing - but wanted to know intuitively why you felt this would not work - since high level algorithm has not changed since your last comment on this.

Reply:
Checked in the basic scenario 

```
it should "correctly print strings after modifier" in {
    class MyModule extends BasicTester {
      val b1 = 10.U
      printf(cf"This is here $b1%x!!!! And should print everything else")
    }
    generateAndCheck(new MyModule,Seq(Printf("This is here %x!!!! And should print everything else",Seq("UInt<4>(\"ha\")"))))

  }
```